src/state/events/logic.js

```diff
@@ -312,25 +288,20 @@ const requestTotalEventsCounts = createLogic({
   processOptions: {
     failType: REQUEST_EVENTS_COUNTS_FAIL,
   },
+  warnTimeout: 0,
   process(deps, dispatch, done) {
     const { firebasedb } = deps;
-    const totalEvents = {
-      pending: 0,
-      live: 0
-    };
-    const p1 = firebasedb.ref(`${EVENTS_PATHS[PENDING_EVENTS_TAB].STATE}`).once('value', (snapshot) => {
+    firebasedb.ref(`${EVENTS_PATHS[PENDING_EVENTS_TAB].STATE}`).on('value', (snapshot) => {
+      let stateEventsCounts = 0;
       if (snapshot.numChildren() > 0) {
         Object.values(snapshot.val()).forEach((state) => {
-          totalEvents.pending += Object.keys(state).length;
+          stateEventsCounts += Object.keys(state).length;
```

Comment:
why arent you no just return snapshot.numbChilrend()? am i missing something here?

Reply:
Yes, that's exactly what I did at first, and then I noticed the count was off because snapshot.numChildren() returns the number of states that have events, not the number of events in those states. The snapshot is at the level of state_legislators_user_submission, e.g.:

state_legislators_user_submission
    [state]
        [event id]
        [event id]
    [state]
        [event id]

...so to get 3 instead of 2 I unfortunately have to iterate through all the states and add those child counts to the total

Reply:
```
snapshot.forEach(stateSnapshot => {
  const count = stateSnapshot.numChildren()
// do whatever adding you're doing. 
})
```

Reply:
Fair enough, didn't realize you could do that with snapshots. Updated...